modules/snowflake-loader/src/test/scala/com/snowplowanalytics/snowplow/loader/snowflake/it/AzureSnowflakeLoaderSpecs.scala

```diff
@@ -0,0 +1,66 @@
+import com.snowplowanalytics.snowplow.rdbloader.common.integrationtestutils.InputBatch
+import com.snowplowanalytics.snowplow.rdbloader.common.integrationtestutils.ItUtils._
+import com.snowplowanalytics.snowplow.rdbloader.experimental.{AzureTestResources, LoaderSpecification}
+
+import org.specs2.matcher.MatchResult
+
+class AzureSnowflakeLoaderSpecs extends LoaderSpecification with SnowflakeTestResources with AzureTestResources {
+  sequential
+
+  "Scenario 1" in {
+    val goodEvent = InputBatch.Content.TextLines(
+      List(
+        """snowplowweb	web	2014-06-01 14:04:11.639	2014-05-29 18:16:35.000	2014-05-29 18:16:35.967	unstruct	2b1b25a4-c0df-4859-8201-cf21492ad61b	114221	clojure	js-2.0.0-M2	clj-0.6.0-tom-0.0.4	hadoop-0.5.0-common-0.4.0		68.42.204.218	1242058182	58df65c46e1ac937	11	437ad25b-2006-455e-b5d8-d664b74df8f3	US	MI	Holland	49423	42.742294	-86.0661						http://snowplowanalytics.com/blog/		https://www.google.com/	http	snowplowanalytics.com	80	/blog/			https	www.google.com	80	/			search	Google							{"schema":"iglu:com.snowplowanalytics.snowplow/contexts/jsonschema/1-0-0","data":[{"schema":"iglu:org.schema/WebPage/jsonschema/1-0-0","data":{"datePublished":"2014-07-23T00:00:00Z","author":"Jonathan Almeida","inLanguage":"en-US","genre":"blog","breadcrumb":["blog","releases"],"keywords":["snowplow","analytics","java","jvm","tracker"]}}]}						{"schema":"iglu:com.snowplowanalytics.snowplow/unstruct_event/jsonschema/1-0-0","data":{"schema":"iglu:com.snowplowanalytics.snowplow/link_click/jsonschema/1-0-0","data":{"targetUrl":"http://snowplowanalytics.com/blog/page2","elementClasses":["next"]}}}																			Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.114 Safari/537.36	Chrome	Chrome		Browser	WEBKIT	en-US	1	1	1	0	1	0	0	0	1	1	24	1241	806	Mac OS	Mac OS	Apple Inc.	America/New_York	Computer	0	1440	900	UTF-8																									"""
+      )
+    )
+    val inputBatches = List(InputBatch(goodEvent))
+    val countExpectations = CountExpectations(good = 1, bad = 0)
+    eventCountCheck(inputBatches, countExpectations)
+  }
+
+  "Scenario 2" in {
+    val countExpectations = CountExpectations(good = 100, bad = 0)
+    val inputBatches = List(InputBatch.good(countExpectations.good))
+    eventCountCheck(inputBatches, countExpectations)
+  }
+
+  def eventCountCheck(inputBatches: List[InputBatch], countExpectations: CountExpectations): MatchResult[Any] = {
+    case class DbActionResult(manifestItems: List[LoaderMessage.ManifestItem], eventIds: List[String])
+    val res = for {
+      (windowAcc, dbActionResult) <- run[DbActionResult](
+                                       inputBatches,
+                                       countExpectations,
+                                       dbActions = testDAO =>
+                                         for {
+                                           manifestItems <- retryUntilNonEmpty(testDAO.queryManifest)
```

Comment:
So we wait until there is any manifest item present in the DB. Is that condition correct when we have multiple windows produced by transformer? 

How about we slightly modify it: we already have here all the messages produced by the transformer. Could we query DB and wait until we find all matching rows for all produced output folders? Try to match  `base` field from shredding_complete message and `base` field from manifest item. For all accumulated messages. Would that be doable? 

Then I think we don't have clean up the table before tests!

Reply:
Oh, good catch! I guess it should be possible to implement it like that. I will give it a shot 👍